File: clientApp/src/GameState/ILayer.ts
import { ITile } from './ITile';
import { IPosition } from './IPosition';
import { IInteractiveObject } from './IInteractiveObject';
import { ICharacter } from './ICharacter';


export interface ILayer {
    getTile(x: number, y: number): ITile;
    getWidth(): number;
    getHeight(): number;
    getSpawn(): IPosition | null;
    getExit(): IPosition | null;
    getInteractiveObjects(): (IInteractiveObject | null)[][];
    getCharacters(): (ICharacter | null)[][];
}

File: ServerApp/ServerApp/GameState/Interfaces/ILayer.cs
﻿using System.Collections.Generic;

namespace ServerApp.GameState
{
    public interface ILayer
    {
        int Height { get; }
        int Width { get; }
        ITile[,] Tiles { get; }
        ICharacter[,] Characters { set; get; }
        IInteractiveObject[,] InteractiveObjects { get; }
        IPosition getExitingPositionOrNull();
        IPosition getEnteringPositionOrNull();
        void AddCharacter(ICharacter character);
        void RemoveCharacterFromPosition(IPosition position);
        ICharacter GetCharacter(IPosition position);
        ITile GetTile(IPosition position);
    }
}